cpp/log/etcd_consistent_store-inl.h

```diff
@@ -544,6 +569,96 @@ void EtcdConsistentStore<Logged>::OnEtcdServingSTHUpdated(
   serving_sth_cv_.notify_all();
 }
 
+
+template <class Logged>
+util::Status EtcdConsistentStore<Logged>::RunOneCleanUpIteration(
+    const int64_t clean_up_to_sequence_number) {
+  CHECK_LE(0, clean_up_to_sequence_number);
+  if (!election_->IsMaster()) {
+    return util::Status(util::error::PERMISSION_DENIED,
+                        "Non-master node cannot run cleanups.");
+  }
+
+  std::vector<EntryHandle<Logged>> sequenced_entries;
+  util::Status status(GetSequencedEntries(&sequenced_entries));
+  if (!status.ok()) {
+    LOG(WARNING) << "Couldn't get sequenced entries: " << status;
+    return status;
+  }
+
+  for (auto& entry : sequenced_entries) {
+    // return if we've hit the limit
+    if (entry.Entry().sequence_number() > clean_up_to_sequence_number) {
+      return util::Status::OK;
+    }
+
+    // Otherwise, first we delete the corresponding unsequenced entry
+    EntryHandle<LoggedCertificate> unseq_entry;
+    status = GetPendingEntryForHash(entry.Entry().Hash(), &unseq_entry);
+    if (!status.ok()) {
+      // Log a warning here, but don't bail on the clean up as we could just be
+      // recovering from a crash halfway through a previous clean up.
+      LOG(WARNING) << "Cleanup couldn't get unsequenced entry " << status;
```

Comment:
What'd be lovely is if the status code got translated to the descriptive string, too :)